api/client/beacon/client_test.go

```diff
@@ -78,3 +79,54 @@ func TestParseNodeVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestValidHostname(t *testing.T) {
+	cases := []struct {
+		name    string
+		hostArg string
+		path    string
+		joined  string
+		err     error
+	}{
+		{
+			name:    "hostname without port",
+			hostArg: "mydomain.org",
+			err:     ErrMalformedHostname,
+		},
+		{
+			name:    "hostname with port",
+			hostArg: "mydomain.org:3500",
```

Comment:
Note that we support `host:port` style addresses and URLs but not a plain hostname. My reasoning for this is that it introduces some base hygeine, so we don't have to worry about what happens when we stuff weird values into `url.URL{Host: ...}`, and we don't need to assume a port without knowing the protocol/scheme.